### PR TITLE
✨ Phenotype model

### PR DESCRIPTION
**Pull request by parimalak**
Added Phenotype model with tests, migrations and mock up data

### CODE REVIEW

File: dataservice/api/phenotype/models.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    :param modified_at: Last time of object modification
+    :param external_id: Name given to Phenotype by contributor
+    :param hpo_id: hpo id
+    :param observed: whether phenotype is negative or positive
+    :param age_at_event_days: Age at the time phenotype was
+            observed, expressed in number of days since birth
+    """
+
+    __tablename__ = 'phenotype'
+    external_id = db.Column(db.Text())
```
Comment (dankolbman) [inline dataservice/api/phenotype/models.py]: What would `external_id` be for a phenotype?

File: migrations/versions/3e79f35badf2_.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+"""empty message
```
Comment (dankolbman) [inline migrations/versions/3e79f35badf2_.py]: Can you add a message here for the log?

File: dataservice/util/data_gen/data_generator.py [hunk 3]
```diff
@@ -143,6 +145,19 @@ def _genomic_files_choices(self):
...
+    def _phenotype_choices(self):
+        """
+        Provides Choices for filling Phenotypes
+        """
+        self.min_phenotypes = 0
+        self.max_phenotypes = 8
+        pref_file = open('dataservice/util/data_gen/phenotype_hpo.csv', 'r')
+        reader = csv.reader(pref_file)
+        self.phenotype_chosen_list = []
+        for row in reader:
```
Comment (dankolbman) [inline dataservice/util/data_gen/data_generator.py:167]: your phenotype csv file only has one line in it. Maybe you wanted some `\n` in there?

File: dataservice/util/data_gen/data_generator.py [hunk 4]
```diff
@@ -310,3 +328,20 @@ def _create_diagnoses(self, total):
...
+    def _create_phenotypes(self, total):
+        """
+        creates phenotypes
+        """
+        phen_list = []
+        for i in range(total):
+            ph = random.choice(self.phenotype_chosen_list)
+            phen = {
+                'external_id': 'phenotype_{}'.format(i),
+                'phenotype': ph[0],
```
Comment (dankolbman) [inline dataservice/util/data_gen/data_generator.py]: `ph[0]` will only get the first character of the string?

File: tests/phenotype/test_phenotype_models.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+from sqlalchemy.exc import IntegrityError
+
+
+class ModelTest(FlaskTestCase):
+	"""
+	Test Phenotype database model
+	"""
+
+	def test_create(self):
+		"""
```
Comment (dankolbman) [inline tests/phenotype/test_phenotype_models.py]: Seems like everything here is overindented.

File: dataservice/api/phenotype/models.py (hunk 1 above)
Comment (parimalak) [reply dataservice/api/phenotype/models.py]: i was thinking to differentiate between multiple phenotypes for the same participant. Seems like kf_id and postgres id are sufficient. i am going to remove it.

File: dataservice/util/data_gen/data_generator.py (hunk 4 above)
Comment (parimalak) [reply dataservice/util/data_gen/data_generator.py]: i am selecting from the list of lists.  
screen shot 2018-02-12 at 1 08 41 pm

File: migrations/versions/3e79f35badf2_.py (hunk 2 above)
Comment (parimalak) [reply migrations/versions/3e79f35badf2_.py]: 👍

File: dataservice/api/phenotype/models.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
+from dataservice.extensions import db
+from dataservice.api.common.model import Base
+
+
+class Phenotype(db.Model, Base):
+    """
+    Phenotype entity.
+    :param _id: Unique id assigned by RDBMS
```
Comment (znatty22) [inline dataservice/api/phenotype/models.py]: Don't need this anymore since we got rid of _id

**Review by znatty22 — COMMENTED** [review]
Looks good, just had a minor comment on models.py and a request..sorry to nitpick here, but could you capitalize the first word in all of your docstrings just to keep things consistent? In some docstrings, the first word is capitalized and in others it isn't

File: dataservice/api/participant/models.py [hunk 7]
```diff
@@ -27,3 +28,7 @@ class Participant(db.Model, Base):
     demographic = db.relationship(Demographic, backref='participant',
                                   uselist=False, cascade="all, delete-orphan",
                                   lazy=True)
+    phenotypes = db.relationship(Phenotype,
+                                 cascade="all, delete-orphan",
+                                 backref=db.backref('participants',
+                                                    lazy=True))
```
Comment (dankolbman) [inline dataservice/api/participant/models.py:34]: Can we set `passive_deletes=True` here to use the db to delete (much faster)

File: dataservice/api/phenotype/models.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+            observed, expressed in number of days since birth
+    """
+
+    __tablename__ = 'phenotype'
+    phenotype = db.Column(db.Text())
+    hpo_id = db.Column(db.Text())
+    observed = db.Column(db.Text())
+    age_at_event_days = db.Column(db.Integer())
+    participant_id = db.Column(db.String(8),
+                               db.ForeignKey('participant.kf_id'),
```
Comment (dankolbman) [inline dataservice/api/phenotype/models.py:24]: And `ondelete='CASCADE'` here for the db. See #122

Comment (parimalak) [reply dataservice/api/phenotype/models.py:24]: does it same for outcome model?

File: dataservice/api/participant/models.py (hunk 7 above)
Comment (parimalak) [reply dataservice/api/participant/models.py:34]: 👍

File: dataservice/api/phenotype/models.py (hunk 8 above)
Comment (dankolbman) [reply dataservice/api/phenotype/models.py:24]: yeah, we should use that for all one-many.

**Review by dankolbman — APPROVED** [review]

**Review by znatty22 — APPROVED** [review]
👍

**Pull request merged**